src/Git/HasTagViaConsole.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\AutomaticReleases\Git;
+
+use function Psl\Shell\execute;
+use function str_contains;
+use function trim;
+
+final class HasTagViaConsole implements HasTag
+{
+    public function __invoke(string $repositoryDirectory, string $tagName): bool
+    {
+        $outout = execute('git', ['tag', '--list', $tagName], $repositoryDirectory);
+
+        if (trim($outout) === '') {
+            return false;
+        }
+
+        return str_contains($outout, $tagName);
```

Comment:
Does `git` not provide an exit code for stuff it doesn't find? 🤔

Reply:
"No", or at least not to my knowledge.

https://git-scm.com/docs/git-tag#Documentation/git-tag.txt---list
https://stackoverflow.com/questions/4917871/does-git-return-specific-return-error-codes